### ISSUE TITLE
Fix PHP Fatal error: Cannot create references to/from string offsets. (date module)

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#4267]
The below was copied from https://github.com/backdrop/backdrop-issues/issues/4027 because I'd like to get the fatal error fixed, and in order to do so we need an issue for the patch :) 

> Today I ran into this issue that has been resolved in the Drupal date module, so I decided to create this meta to get caught up with Date module for Backdrop. I've also filed a PR for only that issue so that I could get my site working again. I hope to have time to address some of these other changes soon too!

Original Drupal issue: https://www.drupal.org/project/date/issues/2449261

**How to reproduce**

- Create a page view of nodes (details don't matter)
- Add an exposed filter on a node date field (like "created")
- Visit that page, enter incomplete data - that's nice, validation strikes
- Now be a hog and override the $_GET values (browser addressbar) with something unusable, like `dateviewsexposed?date_filter[value]foobar`

**You get**

A bunch of errors and `Cannot create references to/from string offsets `.

A screenshot:
![illegal-string-offset](https://user-images.githubusercontent.com/842066/97019876-23144f80-1551-11eb-8b54-405a69568820.png)

---

PR: https://github.com/backdrop/backdrop/pull/2868

**Comment by jenlampton** [comment]
PR updated on Jan 28. Updating issue tags :)

**Comment by docwilmot** [comment]
It would be great to have a more detailed explanation of what this issue is. The D7 issue seems to be about embedded Views in Panel Panes?? We don't have that, so we need some translation to our situation please.

**Comment by docwilmot** [comment]
It would be great to have a more detailed explanation of what this issue is. The D7 issue seems to be about embedded Views in Panel Panes?? We don't have that, so we need some translation to our situation please.

**Comment by indigoxela** [comment]
I agree with @docwilmot - we really need the steps to reproduce this. A short experiment has been successful, though.

**How to reproduce**

- Create a page view of nodes (details don't matter)
- Add an exposed filter on a node date field (like "created")
- Visit that page, enter incomplete data - that's nice, validation strikes
- Now be a hog and override the $_GET values with something unusable, like `dateviewsexposed?date_filter[value]foobar`

**You get**

A bunch of errors and `Cannot create references to/from string offsets `.

A screenshot:
![illegal-string-offset](https://user-images.githubusercontent.com/842066/97019876-23144f80-1551-11eb-8b54-405a69568820.png)

**Comment by indigoxela** [comment]
...garbage, I wanted to import that demo view in the sandbox (for easier reproduction), but the link leads to the installer. The sandbox was too old, I guess.

@jenlampton mind to restart the sandbox?

**Comment by indigoxela** [comment]
@jenlampton many thanks for updating the sandbox.

Without this PR following url would be a breaker:
http://2868.backdrop.backdrop.qa.backdropcms.org/test-date-filter?date_filter[value]foobar

But it isn't in the sandbox.

To compare it with an install without that patch, export the view admin/config/development/configuration/single/export and import it to your test site.

The go to /test-date-filter and add `?date_filter[value]foobar` in your browser's url bar. You'll get "Cannot create references to/from string offsets "

**Comment by indigoxela** [comment]
The PR still looks fine to me, but a [Drupal issue comment](https://www.drupal.org/project/date/issues/2449261#comment-13580621) worries me ("This broke the Search API Views integration with a fatal error").

Search API has been ported to Backdrop, so this change might be a contrib breaker. How can we verify that it isn't?

**Comment by herbdool** [comment]
@indigoxela the search API module made this fix https://git.drupalcode.org/project/search_api/-/commit/a2bdfe7 so should be able to merge this once that's fixed.

**Comment by herbdool** [comment]
I've reported it and likely search API will get a fix soon. I think we can merge this soon.

**Comment by quicksketch** [comment]
We keep bumping this issue to the next release. I believe we're dependent on https://github.com/backdrop-contrib/search_api/issues/24 to be completed first, before this should be merged in core? Or can we merge it at any time?

**Comment by BWPanda** [comment]
Removing from the milestone since this is blocked.

**Comment by laryn** [comment]
I've applied to be a co-maintainer of search_api to help unblock this.

**Comment by laryn** [comment]
I have pulled all the latest commits from the D7 version of `search_api` (including the one @herbdool mentions above) into the Backdrop version. No release yet -- anyone else willing to test?
https://github.com/backdrop-contrib/search_api

**Comment by laryn** [comment]
I am testing this patch on a live site that exhibited this error in the logs periodically related to a particular URL, and it is working well.

**Comment by laryn** [comment]
Thanks to @jenlampton, @klonos, @quicksketch, @indigoxela, and @herbdool for the work on this one, building on the D7 patch by daniel_j, pcho, joseph.olstad, David Hernández, collinhaines, vinmassaro, and SkyNet Costa Rica. (Finally) merged into `1.x` and `1.26.x`.
